libs/admin/src/lib/components/workgroup-edit/workgroup-edit-component.ts

```diff
@@ -1,62 +1,177 @@
+
   public ngOnInit() {
-    this.initializeForm();
-    console.log(this.workgroup);
-  }
-
-  public initializeForm() {
-    this.formGroup.patchValue({
-      name: this.workgroup?.name ?? '',
-      viewers: this.workgroup?.users.filter((user) => user.role === 'Viewer').map((user) => user.userId) ?? [],
-      editors: this.workgroup?.users.filter((user) => user.role === 'Editor').map((user) => user.userId) ?? [],
-      masterEditors:
-        this.workgroup?.users.filter((user) => user.role === 'MasterEditor').map((user) => user.userId) ?? [],
-      status: !!this.workgroup?.disabled_at,
+    this.route.paramMap
+      .pipe(
+        untilDestroyed(this),
+        map((params: ParamMap) => {
+          const id = params.get('id');
+          if (id) {
+            this.mode = 'edit';
+            return this.store.dispatch(actions.findWorkgroup({ workgroupId: parseInt(id) }));
+          } else {
+            this.mode = 'create';
+            this.workgroup = {
+              id: 0,
+              name: '',
+              users: [],
+              assets: [],
+              disabled_at: null,
+            };
+          }
+        })
+      )
+      .subscribe();
+
+    this.initSubscriptions();
+  }
+
+  public ngOnDestroy() {
+    this.store.dispatch(actions.resetWorkgroup());
+  }
+
+  public initializeForm(workgroup: Workgroup) {
+    this.formGroup.patchValue(
+      {
+        name: workgroup.name,
+        status: !!workgroup.disabled_at,
+        users: workgroup.users,
+      },
+      { emitEvent: false }
+    );
+  }
+
+  public updateRoleForUser(event: MatSelectChange, user: UserOnWorkgroup) {
+    if (this.workgroup) {
+      const updatedUsers: UserOnWorkgroup[] = this.workgroup.users.map((u) =>
+        u.userId === user.userId
+          ? {
+              ...u,
+              role: event.value,
+            }
+          : u
+      );
+      const updatedWorkgroup = {
+        name: this.workgroup.name,
+        assets: this.workgroup.assets,
+        disabled_at: this.workgroup.disabled_at,
+        users: updatedUsers,
+      };
+      this.updateWorkgroup(this.workgroup.id, updatedWorkgroup);
+    }
+  }
+
+  public addUsersToWorkgroup() {
+    this.dialogService.open<AddUsersComponent>(AddUsersComponent, {
+      width: '400px',
+      restoreFocus: false,
+      data: {
+        workgroup: this.workgroup,
+        mode: this.mode,
+      },
     });
   }
 
-  public save() {
-    if (!this.formGroup.valid) {
-      return;
+  public deleteUserFromWorkgroup(user: UserOnWorkgroup) {
+    if (this.workgroup) {
+      const updatedWorkgroup = {
+        name: this.workgroup.name,
+        assets: this.workgroup.assets,
+        disabled_at: this.workgroup.disabled_at,
+        users: this.workgroup.users.filter((u) => u.userId !== user.userId),
+      };
+      this.updateWorkgroup(this.workgroup.id, updatedWorkgroup);
+    }
+  }
+
+  public cancel() {
+    this.router.navigate(['../'], { relativeTo: this.route });
```

Comment:
Prefix this call with `void`, as it returns a promise which we want to ignore.
```suggestion
    void this.router.navigate(['../'], { relativeTo: this.route });
```